### PR TITLE
Dynamic slab sizes

### PR DESCRIPTION
**Pull request by replay**
Makes the slab sizes dynamic

also includes a few other minor performance improvements

### CODE REVIEW

**Review by robert-milan — COMMENTED** [review]
Generally I think this will work out quite well. It will be very interesting to start gathering fragmentation and memory usage profiles as we use it more.

I would like to see benchmarks comparing this to master though.

File: slab_pool.go [hunk 1]
```diff
@@ -180,7 +191,9 @@ func (s *slabPool) deleteSlab(slabAddr SlabAddr) (bool, error) {
 	currentSlab := s.slabs[slabIdx]
 
 	// delete slab id from slab slice
-	copy(s.slabs[slabIdx:], s.slabs[slabIdx+1:])
+	for i := slabIdx + 1; i < len(s.slabs); i++ {
```
[the remark this answers is not in the document]
Comment (replay) [reply slab_pool.go]: I have not done any benchmarking with that specific code section. I have done other benchmarks in Metrictank before where I have compared exactly these two alternatives, and when the number of elements to copy was not huge then usually it was faster to stay in user space than having to do a syscall.

File: slab_pool_test.go [hunk 2]
```diff
@@ -45,68 +46,84 @@ func TestAddingDeletingSlabs(t *testing.T) {
 }
 
 func TestAddingGettingManyObjects8(t *testing.T) {
-	testAddingGettingManyObjects(t, 8, 10)
+	testAddingGettingManyObjects(t, 8, 10, 1.3)
```
[the remark this answers is not in the document]
Comment (replay) [reply slab_pool_test.go]: done, gave them better names

**Comment by replay** [comment]
@robert-milan can we merge this?

**Pull request merged**